quinn-proto/src/congestion.rs

```diff
@@ -65,6 +67,10 @@ pub trait Controller: Send + Sync {
     /// Number of ack-eliciting bytes that may be in flight
     fn window(&self) -> u64;
 
+    /// Update implementation-specific metrics.
+    #[cfg(feature = "qlog")]
```

Comment:
I don't think trait methods should be feature-gated: an external congestion controller implementation would break if the feature was enabled on `quinn`. Because `CongestionMetrics` is a simple type we control, there shouldn't be much drawback to exposing this method unconditionally.